core/trino-main/src/main/java/io/trino/operator/scalar/JsonInputFunctions.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.slice.Slice;
+import io.trino.operator.scalar.JsonConversionErrors.JsonInputConversionError;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlType;
+import io.trino.spi.type.StandardTypes;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+
+import static io.trino.json.JsonInputErrorNode.JSON_ERROR;
+import static java.nio.charset.StandardCharsets.UTF_16LE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Read string input as JSON.
+ * <p>
+ * These functions are used by JSON_EXISTS, JSON_VALUE and JSON_QUERY functions
+ * for parsing the JSON input arguments and applicable JSON path parameters.
+ * <p>
+ * If the error handling strategy of the enclosing JSON function is ERROR ON ERROR,
+ * these input functions throw exception in case of parse error.
+ * Otherwise, the parse error is suppressed, and a marker value JSON_ERROR
+ * is returned, so that the enclosing function can handle the error accordingly
+ * to its error handling strategy (e.g. return a default value).
+ */
+public class JsonInputFunctions
+{
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final Charset UTF_32LE = Charset.forName("UTF-32LE");
+
+    private JsonInputFunctions() {}
+
+    @ScalarFunction(value = "$varchar_to_json", hidden = true)
+    @SqlType(StandardTypes.JSON_2016)
+    public static JsonNode varcharToJson(@SqlType(StandardTypes.VARCHAR) Slice inputExpression, @SqlType(StandardTypes.BOOLEAN) boolean failOnError)
+    {
+        Reader reader = new InputStreamReader(inputExpression.getInput(), UTF_8);
+        return toJson(reader, failOnError);
+    }
+
+    @ScalarFunction(value = "$varbinary_to_json", hidden = true)
```

Comment:
why do we need this one if it just redirects to `$varbinary_utf8_to_json`

Reply:
This is intentional:
`FORMAT JSON` without the specified encoding is supposed to use an implementation-dependent encoding. I think that this is the right place to redirect, because it guarantees consistency in the future, when the input functions will be used by other json-related features. And also because the chosen encoding might (unlikely) change at some point.